pkg/generate/git/list_remote_timeout_test.go

```diff
@@ -0,0 +1,47 @@
+package git
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+// TestListRemoteTimeout tests that the `oc new-app` machinery that invokes `git ls-remote`
+// on uncooperative servers correctly times out
+func TestListRemoteTimeout(t *testing.T) {
+	timeout := 1 * time.Millisecond
```

Comment:
not sure this is realistic. It may be the case that it always times out even if you don't sleep on the server handler because of the time it takes to send/receive the request.